### DISCUSSION TITLE
Hasura creates a lot of connection and Postgres CPU utilisation 99.99%

### DISCUSSION THREAD

**Discussion by toton6868** [hasura/graphql-engine#8907] [Question] [answered]
### Version Information

Server Version: 2.7.0
CLI Version (for CLI related issue):

### Environment

Open Source

### What is the current behaviour?

We have created a human tracking application using Android apps. So there is a lot of tracking data is inserting. I have almost 1000 users from where the location tracking data is coming. I am using a database server of 8 Core 16 GB Ram. Now I am facing a problem in App as the data is not inserted. After checking a lot of things I found the CPU utilization of database server is 99.99 or full. My concurrent session in DB is 100. When I list out the sessions I found only the first 7-10 user consumes too much IO read/write and memory usage. Other users are in waiting mode. 

Is there any tuning in Hasura or in Postgres can solve this problem? Any tuning guide link will help.

**Comment by adas98012** [comment, marked as answer]
Hasura does provide some controls over the API Limits (available for cloud and EE users) to prevent long-running GraphQL queries: 

https://hasura.io/docs/latest/security/api-limits/

This may be an option you want to explore.
